timelock-corruption-detection/src/main/java/com/palantir/timelock/history/PaxosLogHistoryProgressTracker.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.history;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.NamespaceAndUseCase;
+import com.palantir.timelock.history.models.ProgressComponents;
+import com.palantir.timelock.history.models.SequenceBounds;
+import com.palantir.timelock.history.sqlite.LogVerificationProgressState;
+import com.palantir.timelock.history.sqlite.SqlitePaxosStateLogHistory;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.sql.DataSource;
+
+public class PaxosLogHistoryProgressTracker {
+    public static final int MAX_ROWS_ALLOWED = 500;
+
+    private final LogVerificationProgressState logVerificationProgressState;
+    private final SqlitePaxosStateLogHistory sqlitePaxosStateLogHistory;
+
+    private Map<NamespaceAndUseCase, ProgressComponents> verificationProgressStateCache = new ConcurrentHashMap<>();
```

Comment:
As far as I can tell, you only ever put or get from this cache (i.e. never delete). I don't have full context on this piece of work, but I wonder if it is possible to OOM yourself? How many entries do you expect to be present in this cache?